File: Python/A_Project/A_Tic_Tac_Toe/GameStateHandler/ConcreteStates/draw_state.py
from GameStateHandler.game_state import GameState


class DrawState(GameState):
    def next(self, context, player, has_won: bool):
        return

    def is_game_over(self) -> bool:
        return True

    def get_result_message(self) -> str:
        return "The game ended in a draw!"

File: Ruby/A_Project/A_Tic_Tac_Toe/GameStateHandler/ConcreteStates/draw_state.rb
require_relative '../game_state'

module GameStateHandler
  module ConcreteStates
    class DrawState
      include GameState

      def next(context, player, has_won)
        # Game is over, no further state transitions
      end

      def game_over?
        true
      end

      def result_message
        "The game ended in a draw!"
      end
    end
  end
end
